contracts/FractalUsul.sol

```diff
@@ -1,86 +1,469 @@
+    /// @param description - proposal description, emitted in ProposalCreated
+    /// @param documentationUrl - proposal documentation/discussion URL, emitted in ProposalCreated
+    function submitProposal(
         address strategy,
         bytes memory data,
         Transaction[] calldata transactions,
         string calldata title,
         string calldata description,
         string calldata documentationUrl
-  ) external {
-      require(
-          isStrategyEnabled(strategy),
-          "voting strategy is not enabled for proposal"
-      );
-      require(transactions.length > 0, "proposal must contain transactions");
-
-      bytes32[] memory txHashes = new bytes32[](transactions.length);
-      for (uint256 i = 0; i < transactions.length; i++) {
-        txHashes[i] = getTransactionHash(
-          transactions[i].to, 
-          transactions[i].value,
-          transactions[i].data, 
-          transactions[i].operation
-        );
-      }
-
-      proposals[totalProposalCount].txHashes = txHashes;
-      proposals[totalProposalCount].strategy = strategy;
-      IStrategy(strategy).receiveProposal(
-          abi.encode(totalProposalCount, txHashes, data)
-      );
-      emit ProposalCreated(strategy, totalProposalCount, msg.sender);
-      emit ProposalMetadataCreated(
-        totalProposalCount, 
-        transactions, 
-        title, 
-        description, 
-        documentationUrl
-      );
-      totalProposalCount++;
-  }
-
-  /// @notice Gets the transaction hashes associated with a given proposald
-  /// @param proposalId The ID of the proposal to get the tx hashes for
-  /// @return bytes32[] The array of tx hashes
-  function getProposalTxHashes(uint256 proposalId) external view returns (bytes32[] memory) {
-    return proposals[proposalId].txHashes;
-  }
-}
\ No newline at end of file
+    ) external {
+        require(
+            isStrategyEnabled(strategy),
+            "Voting strategy is not enabled"
+        );
+        require(transactions.length > 0, "Proposal must contain at least one transaction");
+        require(IBaseStrategy(strategy).isProposer(msg.sender), "Caller cannot submit proposals");
+
+        bytes32[] memory txHashes = new bytes32[](transactions.length);
+        for (uint256 i = 0; i < transactions.length; i++) {
+            txHashes[i] = getTxHash(
+                transactions[i].to,
+                transactions[i].value,
+                transactions[i].data,
+                transactions[i].operation
+            );
+        }
+
+        proposals[totalProposalCount].txHashes = txHashes;
+        proposals[totalProposalCount].strategy = strategy;
+        IBaseStrategy(strategy).receiveProposal(
+            abi.encode(totalProposalCount, txHashes, data)
+        );
+        emit ProposalCreated(
+            strategy,
+            totalProposalCount,
+            msg.sender,
+            transactions,
+            title,
```

Comment:
@tbwebb22 is it worth changing the contract to take a "metadata" string now, instead of three strings

then the metadata string we give it can be a JSON string with title/description/url instead and the frontend gets that data from the event

in a future update we can send an IPFS hash for the metadata, and achieve what you're saying without contract changes?

Reply:
Yeah for sure, I like that. Just a single metadata string makes sense

Reply:
nice, yeah be sure to add that to the list of frontend changes